url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+                     //que unen la casilla del rey y la casilla de la pieza
+                     //Si sí que hay alguna pieza se generan los movimientos de la pieza de forma normal
+                     bbConnection = bbExclusiveConnection[pieceSquareIndex][kingSquareIndex];
+                     if((bbConnection & bbOccupation) == 0)
+                     {
+                        //Si no hay casillas se comprueba si hay alguna pieza que ataque a la pieza en la misma
+                        //linea que el rey y la pieza
+                        //Si no hay ninguna se generan los movimientos de la pieza de forma normal
+                        bbAttacker = bbFullConnection[pieceSquareIndex][kingSquareIndex] & 
+                              bbAttackerToSquare[pieceSquareIndex] & bbOppositeOccupied;
+                        if(bbAttacker != 0)
+                        {
+                           genericPiece = pieceInSquare[Long.numberOfTrailingZeros(bbAttacker)].genericPiece;
+                           if(genericPiece != GenericPiece.PAWN && genericPiece != GenericPiece.KING)
+                           {
+                              //Si hay alguna pieza atacando en esa linea quiere decir que la pieza
+                              //sólo se podrá mover a las casillas que conectan las casilla de la pieza
+                              //y la casilla del atacante.
+                              isPinned = true;
+                           }
+                        }
+                     }
+                  }
+
+                  if(isPinned == false)
+                  {
+                     if(attackedSquareIndex < 56 && attackedSquareIndex > 7)
+                     {
+                        mov = new Movement(square[pieceSquareIndex], square[attackedSquareIndex], Piece.WHITE_PAWN);
+
+                        movesList.add(mov);
+                     }
+                     else
+                     {
+                        mov = new Movement(square[pieceSquareIndex], square[attackedSquareIndex], Piece.WHITE_PAWN, null, Piece.WHITE_ROOK);
+
+                        movesList.add(mov);
+                        
+                        mov = new Movement(square[pieceSquareIndex], square[attackedSquareIndex], Piece.WHITE_PAWN, null, Piece.WHITE_KNIGHT);
+
+                        movesList.add(mov);
+                        
+                        mov = new Movement(square[pieceSquareIndex], square[attackedSquareIndex], Piece.WHITE_PAWN, null, Piece.WHITE_BISHOP);
+
+                        movesList.add(mov);
+                        
+                        mov = new Movement(square[pieceSquareIndex], square[attackedSquareIndex], Piece.WHITE_PAWN, null, Piece.WHITE_QUEEN);
+
+                        movesList.add(mov);
+                     }
+                  }
+                  
+                  advances1 ^= bbSquare[attackedSquareIndex];
+               }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 2035:2103, 2166:2234.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.